## Instructions {-}

1. Please answer the following questions as part of your project proposal.

2. Write your answers in the *Markdown* cells of the Jupyter notebook. You don't need to write any code, but if you want to, you may use the *Code* cells.

3. Use [Quarto](https://quarto.org/docs/output-formats/html-basics.html) to print the *.ipynb* file as HTML. You will need to open the command prompt, navigate to the directory containing the file, and use the command: `quarto render filename.ipynb --to html`. Submit the HTML file.

4. The project proposal is worth 8 points, and is due on **17th October 2023 at 11:59 pm**. 

5. You must make one submission as a group, and not individually.

6. Maintaining a GitHub repository is optional, though encouraged for the project.

7. Share the link of your project’s GitHub repository on the [spreadsheet](https://docs.google.com/spreadsheets/d/1khao3unpj_vsx4kOSg_Zzo77YK1UWL2w73Oa0aAirOo/edit#gid=0).

# 1) Team name
Mention your team name.

*(0 points)*

Womp

# 2) Member names
Mention the names of your team members.



*(0 points)*

Yuki Han,	
Fanyi Mo,	
Antonio Santa Cruz,	
Vincent Zhang

# 3) Topic
Mention the topic of your course project.

*(0 points)*

Understanding the interplay of hospital treatments (including diagnosis, hospital-administered tests and drugs) and various patient data points in the context of diabetes care.

# 4) Inspiration
What inspired your team to choose this topic?

*(0.1 points)*

We have a general interest in using data to generate better healthcare outcomes. The medical field can benefit from the determination of trends in medical procedure and outcomes that can not be easily understood without large aggregate data.

# 5) EDA (Exploratory data analysis) questions

Explain the questions to be answered in your course project. You must have 4 questions - one for each team member. Mention the name of the team member and the question that they will answer against it.

*(0.4 points)*

1. Readmission Rate: How the various drugs taken by the patient and the type of care received influences readmission rates? (Vincent Zhang)

2. Length of Stay: How might shorter or longer hospital stays enhance the patient outcomes and lower the burden on healthcare resources? (Fanyi Mo)

3. Number of Emergency Visits in the Year Preceding the Encounter: How is the number of emergency visits the diabetic patients have in the year leading up to their hospital encounter related to the probability of a successful inpatient treatment for the patient? (Yuki Han)

4. Number of Procedures (Other than Lab Tests) Performed During the Encounter: Can we predict the successful recovery rate of a patient based on the number procedures they had during their stay? (Antonio Santa Cruz)



# 6) Data 

What data sources will you use, and how will the data help answer the questions? Explain.

If the data is open source, share the link of the data.

What is the number of continuous variables, categorical variables, and observations in your dataset(s). If you are using multiple datasets, please provide the information for each dataset. When counting variables, count only those that have sufficient non-missing values, and will be useful.

Which variable is the main variable of interest? For example, if you are analyzing election trends, then the variable consisting of number of votes may be the main variable of interest. If you are analyzing crime rate, the variable consisting of the number of crimes may be the main variable of interest.

*Note: You cannot use any of the datasets mentioned [here](https://sites.google.com/view/stat303-1fall2023/project).*

*(0.4 points)*

In [2]:
! pip uninstall ucimlrepo -y

In [3]:
from ucimlrepo import fetch_ucirepo

diabetes_130_us_hospitals_for_years_1999_2008 = fetch_ucirepo(id=296)
X = diabetes_130_us_hospitals_for_years_1999_2008.data.features 
y = diabetes_130_us_hospitals_for_years_1999_2008.data.targets

print(diabetes_130_us_hospitals_for_years_1999_2008.metadata)
print(diabetes_130_us_hospitals_for_years_1999_2008.variables)

{'uci_id': 296, 'name': 'Diabetes 130-US hospitals for years 1999-2008', 'repository_url': 'https://archive.ics.uci.edu/dataset/296/diabetes+130-us+hospitals+for+years+1999-2008', 'data_url': 'https://archive.ics.uci.edu/static/public/296/data.csv', 'abstract': 'The dataset represents ten years (1999-2008) of clinical care at 130 US hospitals and integrated delivery networks. Each row concerns hospital records of patients diagnosed with diabetes, who underwent laboratory, medications, and stayed up to 14 days. The goal is to determine the early readmission of the patient within 30 days of discharge.\nThe problem is important for the following reasons. Despite high-quality evidence showing improved clinical outcomes for diabetic patients who receive various preventive and therapeutic interventions, many patients do not receive them. This can be partially attributed to arbitrary diabetes management in hospital environments, which fail to attend to glycemic control. Failure to provide pro

/Users/hanziyan/Desktop/STAT303_GROUP/ucimlrepo/fetch.py:100: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(data_url)


In [4]:
ids = diabetes_130_us_hospitals_for_years_1999_2008.data.ids
data = diabetes_130_us_hospitals_for_years_1999_2008.data.features

data = data.join(ids)

data["readmitted"] = diabetes_130_us_hospitals_for_years_1999_2008.data.targets

data = data.drop_duplicates(subset=["patient_nbr"], keep='first')

data = data.loc[data.gender!="Unknown/Invalid",:]


----------------------------------------------------------------
Link to dataset: https://archive.ics.uci.edu/dataset/296/diabetes+130-us+hospitals+for+years+1999-2008

The dataset represents ten years (1999-2008) of clinical care at 130 US hospitals and integrated delivery networks. Each row concerns hospital records of patients diagnosed with diabetes, who underwent laboratory, medications, and stayed up to 14 days. The data contains such attributes as patient number, race, gender, age, admission type, time in hospital, medical specialty of admitting physician, number of lab tests performed, HbA1c test result, diagnosis, number of medications, diabetic medications, number of outpatient, inpatient, and emergency visits in the year before the hospitalization, etc.

Number of Observations = 101766
Number of Categorical Variables = 40
Number of Continuous Variables = 8

Variables of Interest = time_in_hospital, readmitted, number_emergency, num_procedures, num_medications, and all drug variables

# 7) Stakeholders

Who are the stakeholders, and how will your project benefit them? Explain.

*(0.1 points)*

The stakeholders a healthcare providers, hospital administrators, as well as patients and their families. 

This research project centers around the interplay of hospital treatments (including diagnosis, hospital-administered tests and drugs) and various patient data points in the context of diabetes care. The result of our study holds significant promise for the stakeholders of the Northwestern Medicine (NM) ecosystem: healthcare providers and hospitals under NM can better understand the relationships between factors like readmission rates, length of stay, prior emergency visits, and procedural interventions that can lead to more informed and tailored treatment decisions. This, in turn, enhances patient outcomes, reduces healthcare costs, and optimizes resource allocation. For example, if it is found that some drugs are less effective or rarely prescribed, we may suggest that hospitals stockpile less of these drugs.  